### ISSUE TITLE
Android custom marker incorrect anchor point

### ISSUE THREAD

**Issue by levic92** [lelandrichardson/react-native-maps#230]
Hello,

I have a map with custom marker points on it and they are no longer in the correct position on the map.

### Correct
![10](https://cloud.githubusercontent.com/assets/1757742/15259472/5223450c-1906-11e6-95f4-d85b0dc64b23.jpg)
### Incorrect
![11](https://cloud.githubusercontent.com/assets/1757742/15259475/5561be7e-1906-11e6-8890-683977955050.jpg)

Earlier in the week the position of the markers were all correct (1st image), then on Wednesday I noticed that the markers were in a different location and on zoom in and out they moved around weird, I went back to previous commits of my code but nothing resolved the issue. I believe this issue came up from a google play services update that occurred on Tuesday.

I can somewhat fix the problem by setting an anchor point on the marker, the problem with that is only android v4.4.2 and down are affected by the issue and the anchor points vary from device to device. (there is no problem on 4.4.4+ and iOS)

I believe what is happening is that the custom marker bitmap size is larger than the actual view. The view itself always looks correct (26x26) but the map is behaving like the image is double that size, so the anchor point is completely off.

Any help on this issue would be greatly appreciated,

code:
```js
class CustomMarker extends React.Component {
  render() {
    return (
      <View style={styles.markerWrapper}>
        <Text style={styles.markerText}>{this.props.markerNumber}</Text>
      </View>
    )
  }
}
//...
renderMarker(i) {
    let markerNumber = i+1;
    return (
      <MapView.Marker
        //...
        coordinate={{
          latitude: lat,
          longitude: lng,
        }}>
        <CustomMarker markerNumber={markerNumber} />
      </MapView.Marker>
    );
  }
//...
let styles = StyleSheet.create({
  markerWrapper: {
    borderRadius: 13,
    width: 26,
    height: 26,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: cssVar("brandColor"),
  },
  markerText: {
    color: "#FFF",
    fontWeight: "700",
  },
});
```

note: I have also tried a simple square view with just a background color and the problem still persists.

**Comment by Thetortuga** [comment]
Hello, 

By reading your issue, I have the feeling that you fell into the same issue than me : #228 
I am happy to see that I am not alone with it !

**Comment by mvaivre** [comment]
Same problem here. Last week everything was ok, and today the bug appears on my device but not on my simulator (where updates are not done automatically).

**Comment by jonesdar** [comment]
+ 1 I've also noticed that when rotating the map the markers now follow the map rotation - so end up upside-down after a 180° map rotation. Previously they would stay upright as expected.

**Comment by mvaivre** [comment]
@jonesdar Actually it would be great to get the possibility to specify if we want the marker to float over the map or to follow the rotation (I would need to use the later for my current project).

**Comment by jonesdar** [comment]
More info at https://code.google.com/p/gmaps-api-issues/issues/detail?id=9768

**Comment by jonesdar** [comment]
The workaround described in the google issue fixes this problem - in AirMapMarker.update() add a setAnchor() call after the setIcon() call:

    public void update() {
        if (marker == null) {
            return;
        }

        marker.setIcon(getIcon());
    +   if (anchorIsSet) {
    +       marker.setAnchor(anchorX, anchorY);
    +   }
    }

They also recommend doing the same for setInfoWindowAnchor().

**Comment by jrichardlai** [comment]
Awesome @jonesdar can you make a PR for it?

**Comment by levic92** [comment]
PR has been created. Thanks @jonesdar for finding the solution.

**Comment by osungjin** [comment]
@levic92 Thanks

**Comment by mvaivre** [comment]
As stated here https://github.com/lelandrichardson/react-native-maps/pull/258, this fix is not working on my side. The behavior is pretty random, the anchor position can be different after a full reload, but correct after a hot reload. Maybe I don't set the custom marker height correctly, so the relative anchor (0.5, 0.5) doesn't work well? I use paddings and the content (text), can differ in size... 

Is it working on your side guys? I need to go to production in the following days, so I really need to find a way to fix that.

**Comment by mvaivre** [comment]
Update: I tested the change manually with ver. 0.4.2 and it works.
So the problem seems to be related to the package linking for ver. 0.5, the changes I made in java files were not taken into account.

**Comment by DennisMG** [comment]
I'm experiencing the same issue as @mvaivre .. The fix does not work with ver 0.5..  changing to ver 0.4.2 for the meantime..
